compiler/src/formatting/fmt.re

```diff
@@ -0,0 +1,3252 @@
+              ~lead=space,
+              ~trail=space,
+              true_branch.pexp_loc,
+              Option.get(false_branch).pexp_loc,
+            )
+            ++ string("else ")
+            ++ false_branch_doc
+          | None => empty
+          }
+        ),
+      );
+    } else {
+      switch (true_branch.pexp_desc, false_branch) {
+      | (PExpBlock(_), _)
+      | (_, Some({pexp_desc: PExpBlock(_) | PExpIf(_)})) =>
+        print_if(
+          ~loc,
+          ~force_blocks=true,
+          condition,
+          true_branch,
+          false_branch,
+        )
+      | (_, None) =>
+        let true_branch_doc =
+          switch (true_branch.pexp_desc) {
+          | PExpIf(_) => parens(print_expression(true_branch))
+          | _ => print_expression(true_branch)
+          };
+        group(
+          string("if ")
+          ++ parens(
+               print_comment_range(
+                 ~block_start=true,
+                 ~trail=space,
+                 enclosing_start_location(loc),
+                 condition.pexp_loc,
+               )
+               ++ print_expression(~infix_wrap=Fun.id, condition)
+               ++ print_comment_range(
+                    ~block_end=true,
+                    ~lead=space,
+                    condition.pexp_loc,
+                    true_branch.pexp_loc,
+                  ),
+             )
+          ++ space
+          ++ true_branch_doc,
+        );
+      | (_, Some(false_branch)) =>
+        let true_branch_doc =
+          switch (true_branch.pexp_desc) {
+          | PExpIf(_) => parens(print_expression(true_branch))
+          | _ => print_expression(true_branch)
+          };
+        group(
+          string("if ")
+          ++ parens(
+               print_comment_range(
+                 ~block_start=true,
+                 ~trail=space,
+                 enclosing_start_location(loc),
+                 condition.pexp_loc,
+               )
+               ++ print_expression(~infix_wrap=Fun.id, condition)
+               ++ print_comment_range(
+                    ~block_end=true,
+                    ~lead=space,
+                    condition.pexp_loc,
+                    true_branch.pexp_loc,
+                  ),
+             )
+          ++ indent(2, breakable_space ++ true_branch_doc)
+          ++ print_comment_range(
+               ~none=breakable_space,
+               ~lead=space,
+               ~trail=breakable_space,
+               true_branch.pexp_loc,
+               false_branch.pexp_loc,
+             )
+          ++ string("else")
+          ++ indent(2, breakable_space ++ print_expression(false_branch)),
+        );
+      };
+    }
+  and print_assignment = (~collapsible, ~lhs_loc, new_value) => {
+    switch (new_value.pexp_desc) {
+    | PExpApp(
+        {
+          pexp_desc:
+            PExpId({
+              txt:
+                Identifier.IdentName({
+                  txt: ("+" | "-" | "*" | "/" | "%") as op,
```

Comment:
Why do we only want to treat this particular set differently?

Reply:
These are the only ones that Grain allows with the assignment syntax.